File: RegularExpressionForOc/RegularExpressionForOc/ViewController.m
//
//  ViewController.m
//  RegularExpressionForOc
//
//  Created by Mac on 16/5/16.
//  Copyright © 2016年 zhanglei. All rights reserved.
//
#define SCREEN_WIDTH [UIScreen mainScreen].bounds.size.width
#define SCREEN_HEIGHT [UIScreen mainScreen].bounds.size.height
#define TABLEVIEWCELL_IDENT @"tableviewcell_ident"
#import "ViewController.h"
#import "RegularExpressionTool.h"
#import "Toast+UIView.h"
@interface ViewController ()
{
    UITableView * regularTableView;
    NSMutableArray * holderArray;
    NSMutableDictionary * utfDic;
}
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
 
    [self initData];
    
    [self.view setBackgroundColor:[UIColor redColor]];
    
    UILabel * lblTitle = [[UILabel alloc] initWithFrame:CGRectMake(0, 0, SCREEN_WIDTH, 60)];
    [lblTitle setText:@"正则验证"];
    [lblTitle setTextAlignment:NSTextAlignmentCenter];
    [lblTitle setTextColor:[UIColor whiteColor]];
    [lblTitle setBackgroundColor:[UIColor blackColor]];
    [self.view addSubview:lblTitle];
    
    regularTableView = [[UITableView alloc] initWithFrame:CGRectMake(0, 60, SCREEN_WIDTH, SCREEN_HEIGHT-60)];
    regularTableView.separatorStyle = UITableViewCellSeparatorStyleNone;
    regularTableView.dataSource = self;
    regularTableView.delegate = self;
    regularTableView.backgroundColor = [UIColor whiteColor];
    [regularTableView.layer setBorderColor:[UIColor whiteColor].CGColor];
    [self.view addSubview:regularTableView];
    
}

-(void)initData{
    NSArray * arr = @[@"账号",@"密码",@"手机号",@"邮箱",@"中文2~4",@"验证数字",@"日期",@"金额"];
    holderArray = [NSMutableArray arrayWithArray:arr];
    utfDic = [[NSMutableDictionary alloc] init];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

-(CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath{
    return 40;
}

-(UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath{
    UITableViewCell * cell = [tableView dequeueReusableCellWithIdentifier:TABLEVIEWCELL_IDENT];
    if (!cell) {
        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:TABLEVIEWCELL_IDENT];
    }
    
    UITextField * utf = [[UITextField alloc] initWithFrame:CGRectMake(5, 0, tableView.frame.size.width-60, 40)];
    utf.delegate = self;
    [utf.layer setMasksToBounds:YES];
    [utf.layer setCornerRadius:4];
    utf.placeholder = [holderArray objectAtIndex:indexPath.row];
    [cell addSubview:utf];
    
    [utfDic setObject:utf forKey:[NSString stringWithFormat:@"%ld",indexPath.row]];
    
    UIButton * btn = [[UIButton alloc] initWithFrame:CGRectMake(tableView.frame.size.width-45, 0, 40, 40)];
    [btn setTag:indexPath.row];
    [btn addTarget:self action:@selector(validate:) forControlEvents:UIControlEventTouchUpInside];
    [btn setBackgroundColor:[UIColor blackColor]];
    [btn setTitle:@"验证" forState:UIControlStateNormal];
    [cell addSubview:btn];
    
    UIView * line = [[UIView alloc] initWithFrame:CGRectMake(0, 39.5, SCREEN_WIDTH, .5)];
    [line setBackgroundColor:[UIColor grayColor]];
    [cell addSubview:line];
    
    cell.selectionStyle = UITableViewCellSelectionStyleNone;
    return cell;
}

-(NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section{
    return holderArray.count;
}

-(void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath{
    
}

-(void)validate:(id)sender{
    UIButton * button = (UIButton *)sender;
    NSInteger tag = button.tag;
    RegularExpressionTool * tool  =[RegularExpressionTool sharedManager];
    switch (tag) {
        case 0:
        {
            UITextField * utf = [utfDic objectForKey:[NSString stringWithFormat:@"%ld",tag]];
            BOOL isTrue = [tool validateAccount:utf.text];
            if (isTrue) {
                [self.view makeToast:@"账号信息正确" withValue:@"" withTextColor:[UIColor greenColor]];
            }else{
                [self.view makeToast:@"账号信息错误" withValue:@"" withTextColor:[UIColor redColor]];
            }
        }
            break;
        case 1:
        {
            UITextField * utf = [utfDic objectForKey:[NSString stringWithFormat:@"%ld",tag]];
            BOOL isTrue = [tool validatePassword:utf.text];
            if (isTrue) {
                [self.view makeToast:@"密码信息正确" withValue:@"" withTextColor:[UIColor greenColor]];
            }else{
                [self.view makeToast:@"密码信息错误" withValue:@"" withTextColor:[UIColor redColor]];
            }
        }
            break;
        case 2:
        {
            UITextField * utf = [utfDic objectForKey:[NSString stringWithFormat:@"%ld",tag]];
            BOOL isTrue = [tool validatePhone:utf.text];
            if (isTrue) {
                [self.view makeToast:@"手机号码信息正确" withValue:@"" withTextColor:[UIColor greenColor]];
            }else{
                [self.view makeToast:@"手机号码信息错误" withValue:@"" withTextColor:[UIColor redColor]];
            }
        }
            break;
        case 3:
        {
            UITextField * utf = [utfDic objectForKey:[NSString stringWithFormat:@"%ld",tag]];
            BOOL isTrue = [tool validateEmail:utf.text];
            if (isTrue) {
                [self.view makeToast:@"邮箱信息正确" withValue:@"" withTextColor:[UIColor greenColor]];
            }else{
                [self.view makeToast:@"邮箱信息错误" withValue:@"" withTextColor:[UIColor redColor]];
            }
        }
            break;
        case 4:
        {
            UITextField * utf = [utfDic objectForKey:[NSString stringWithFormat:@"%ld",tag]];
            BOOL isTrue = [tool validateChinese2_4:utf.text];
            if (isTrue) {
                [self.view makeToast:@"中文昵称信息正确" withValue:@"" withTextColor:[UIColor greenColor]];
            }else{
                [self.view makeToast:@"中文昵称信息错误" withValue:@"" withTextColor:[UIColor redColor]];
            }
        }
            break;
        case 5:
        {
            UITextField * utf = [utfDic objectForKey:[NSString stringWithFormat:@"%ld",tag]];
            BOOL isTrue = [tool validateNumber:utf.text];
            if (isTrue) {
                [self.view makeToast:@"数字信息正确" withValue:@"" withTextColor:[UIColor greenColor]];
            }else{
                [self.view makeToast:@"数字信息错误" withValue:@"" withTextColor:[UIColor redColor]];
            }
        }
            break;
        case 6:
        {
            UITextField * utf = [utfDic objectForKey:[NSString stringWithFormat:@"%ld",tag]];
            BOOL isTrue = [tool validateDate:utf.text];
            if (isTrue) {
                [self.view makeToast:@"日期信息正确" withValue:@"" withTextColor:[UIColor greenColor]];
            }else{
                [self.view makeToast:@"日期信息错误" withValue:@"" withTextColor:[UIColor redColor]];
            }
        }
            break;
        case 7:
        {
            UITextField * utf = [utfDic objectForKey:[NSString stringWithFormat:@"%ld",tag]];
            BOOL isTrue = [tool validatePrice:utf.text];
            if (isTrue) {
                [self.view makeToast:@"金额信息正确" withValue:@"" withTextColor:[UIColor greenColor]];
            }else{
                [self.view makeToast:@"金额信息错误" withValue:@"" withTextColor:[UIColor redColor]];
            }
        }
            break;
            
        default:
            break;
    }
}

@end

File: RegularExpressionForSwift/RegularExpressionForSwift/ViewController.swift
//
//  ViewController.swift
//  RegularExpressionForSwift
//
//  Created by Mac on 16/5/16.
//  Copyright © 2016年 zhanglei. All rights reserved.
//

import UIKit

let SCREEN_WIDTH = UIScreen.mainScreen().bounds.size.width
let SCREEN_HEIGHT = UIScreen.mainScreen().bounds.size.height

class ViewController: UIViewController,UITableViewDelegate,UITableViewDataSource {
    
    var regularEx : RegularExpressionTool!
    var tview : Toast_UIView!
    var tbView : UITableView!
    var validateArray : [String] = ["账号","密码","手机号","邮箱","中文2~4","验证数字","日期","金额"]
    var dic:NSMutableDictionary = [:]
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a     nib.
        
        regularEx = RegularExpressionTool();
        tview = Toast_UIView();
        tview.backgroundColor = UIColor.whiteColor()
        self.view = tview;
        
        let label: UILabel = UILabel(frame: CGRectMake(0, 0, SCREEN_WIDTH, 60))
        label.textAlignment = NSTextAlignment.Center
        label.textColor = UIColor.whiteColor()
        label.backgroundColor = UIColor.blackColor()
        label.text = "正则验证"
        self.view.addSubview(label)
        
        tbView = UITableView(frame: CGRectMake(0, 60, SCREEN_WIDTH, SCREEN_HEIGHT-60))
        tbView.backgroundColor = UIColor.whiteColor()
        tbView.dataSource = self
        tbView.delegate = self
        
        self.view.addSubview(tbView)
        
//        let b = regularEx.validateAccount("aaaa");
//        if b == true {
//            print("account is available");
//            tview.makeToast("account is available", value: "", color: UIColor.greenColor())
//        }else{
//            print("account is not available");
//            tview.makeToast("account is not available", value: "", color: UIColor.redColor())
//        }
        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    func tableView(tableView: UITableView, heightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat {
        return 40;
    }
    
    func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
        
    }
    
    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        let CellIdent: String = "cellident"
        var tableCell: UITableViewCell? = tableView.dequeueReusableCellWithIdentifier(CellIdent)
        
        if tableCell == nil {
            tableCell = UITableViewCell(style: UITableViewCellStyle.Default, reuseIdentifier: CellIdent)
        }
        
        let frame = CGRectMake(5, 0, tableView.frame.size.width-60, 40)
        let utf = UITextField(frame: frame)
        utf.tag = indexPath.row
        utf.placeholder = validateArray[indexPath.row]
        dic.setObject(utf, forKey: NSString(format:"%d",indexPath.row ))
        tableCell?.addSubview(utf)
        
        let button = UIButton(frame:CGRectMake(tableView.frame.size.width-50, 0, 40, 40))
        button.tag = indexPath.row
        button.backgroundColor = UIColor.redColor()
        button.setTitle("验证", forState: UIControlState.Normal)
        button.setTitleColor(UIColor.blackColor(), forState: UIControlState.Normal)
        button.addTarget(self, action:#selector(ViewController.validate(_:)) , forControlEvents: UIControlEvents.TouchUpInside)
        tableCell?.addSubview(button)
        
        let line:UIView = UIView(frame: CGRectMake(0, 39.5, SCREEN_WIDTH, 0.5))
        line.backgroundColor = UIColor.grayColor()
        tableCell?.addSubview(line)
        tableCell?.selectionStyle = UITableViewCellSelectionStyle.None
        return tableCell!
    }
    
    func validate(uibutton:UIButton) -> Void {
        switch uibutton.tag {
        case 0:
            let textField = dic[NSString(format:"%d",uibutton.tag)] as! UITextField
            let b = regularEx.validateAccount(textField.text!)
            if b == true {
                 tview.makeToast("account is available", value: "", color: UIColor.greenColor())
            }else{
                 tview.makeToast("account is not available", value: "", color: UIColor.redColor())
            }
            break;
        case 1:
            let textField = dic[NSString(format:"%d",uibutton.tag)] as! UITextField
            let b = regularEx.validatePassword(textField.text!)
            if b == true {
                tview.makeToast("password is available", value: "", color: UIColor.greenColor())
            }else{
                tview.makeToast("password is not available", value: "", color: UIColor.redColor())
            }
            break;
        case 2:
            let textField = dic[NSString(format:"%d",uibutton.tag)] as! UITextField
            let b = regularEx.validatePassword(textField.text!)
            if b == true {
                tview.makeToast("password is available", value: "", color: UIColor.greenColor())
            }else{
                tview.makeToast("password is not available", value: "", color: UIColor.redColor())
            }
            break;
        case 3:
            let textField = dic[NSString(format:"%d",uibutton.tag)] as! UITextField
            let b = regularEx.validatePhone(textField.text!)
            if b == true {
                tview.makeToast("phone is available", value: "", color: UIColor.greenColor())
            }else{
                tview.makeToast("phone is not available", value: "", color: UIColor.redColor())
            }
            break;
        case 4:
            let textField = dic[NSString(format:"%d",uibutton.tag)] as! UITextField
            let b = regularEx.validateEmail(textField.text!)
            if b == true {
                tview.makeToast("email is available", value: "", color: UIColor.greenColor())
            }else{
                tview.makeToast("email is not available", value: "", color: UIColor.redColor())
            }
            break;
        case 5:
            let textField = dic[NSString(format:"%d",uibutton.tag)] as! UITextField
            let b = regularEx.validateChineseName(textField.text!)
            if b == true {
                tview.makeToast("chineseName is available", value: "", color: UIColor.greenColor())
            }else{
                tview.makeToast("chineseName is not available", value: "", color: UIColor.redColor())
            }
            break;
        case 6:
            let textField = dic[NSString(format:"%d",uibutton.tag)] as! UITextField
            let b = regularEx.validateNumber(textField.text!)
            if b == true {
                tview.makeToast("number is available", value: "", color: UIColor.greenColor())
            }else{
                tview.makeToast("number is not available", value: "", color: UIColor.redColor())
            }
            break;
        case 7:
            let textField = dic[NSString(format:"%d",uibutton.tag)] as! UITextField
            let b = regularEx.validateDate(textField.text!)
            if b == true {
                tview.makeToast("date is available", value: "", color: UIColor.greenColor())
            }else{
                tview.makeToast("date is not available", value: "", color: UIColor.redColor())
            }
            break;
        case 8:
            let textField = dic[NSString(format:"%d",uibutton.tag)] as! UITextField
            let b = regularEx.validatePrice(textField.text!)
            if b == true {
                tview.makeToast("price is available", value: "", color: UIColor.greenColor())
            }else{
                tview.makeToast("price is not available", value: "", color: UIColor.redColor())
            }
            break;
            
        default:
            break;
        }
    }
    
    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return validateArray.count;
    }

}

